flow/src/main/java/com/sequenceiq/flow/service/FlowService.java

```diff
@@ -260,4 +266,33 @@ public Optional<OperationFlowsView> getLastFlowOperationByResourceCrn(String res
         List<FlowLog> flowLogs = flowLogDBService.getLatestFlowLogsByCrnInFlowChain(resourceCrn);
         return flowOperationStatisticsService.createOperationResponse(resourceCrn, flowLogs);
     }
+
+    public List<FlowLogResponse> getFlowLogsByIds(List<String> flowIds, Pageable pageable) {
+        LOGGER.info("Getting flow logs by flow ids list {}", flowIds);
+        return flowLogDBService.getFlowLogsByFlowIdsCreatedDesc(new HashSet<>(flowIds), pageable)
+                .stream().map(flowLog -> flowLogConverter.convert(flowLog)).collect(toList());
+    }
+
+    public List<FlowCheckResponse> getFlowChainsByChainIds(List<String> chainIds, Pageable pageable) {
+        LOGGER.info("Getting flow check responses by flow check ids list {}", chainIds);
+        Map<String, FlowChainLog> flowChainsMap = flowChainLogService
+                .findAllByFlowChainIdInOrderByCreatedDesc(new HashSet<>(chainIds), pageable)
+                .stream().collect(Collectors.toMap(FlowChainLog::getFlowChainId, Function.identity()));
+        Set<String> flowChainIds = new HashSet<>(flowChainsMap.keySet());
+        Map<String, List<FlowLogWithoutPayload>> flowLogsByChainId = flowLogDBService
+                .getFlowLogsWithoutPayloadByFlowChainIdsCreatedDesc(flowChainIds)
+                .stream().collect(groupingBy(FlowLogWithoutPayload::getFlowChainId, LinkedHashMap::new, toList()));
+        return flowChainIds.stream().map(key -> convertFlowChainToFlowCheckResponse(flowChainsMap.get(key),
+                flowLogsByChainId.getOrDefault(key, List.of()))).collect(toList());
+    }
+
+    private FlowCheckResponse convertFlowChainToFlowCheckResponse(FlowChainLog chainLog, List<FlowLogWithoutPayload> relatedFlowLogs) {
+        FlowCheckResponse flowCheckResponse = new FlowCheckResponse();
+        flowCheckResponse.setFlowChainId(chainLog.getFlowChainId());
+        flowCheckResponse.setHasActiveFlow(!completed("Flow chain", chainLog.getFlowChainId(), List.of(chainLog),
+                relatedFlowLogs));
+        flowCheckResponse.setLatestFlowFinalizedAndFailed(isFlowInFailedState(relatedFlowLogs, failHandledEvents));
+        setEndTimeOnFlowCheckResponse(flowCheckResponse, relatedFlowLogs);
```

Comment:
nit: I'm not requesting a change here as this is already merged code, but it would be better if this method would return with a time (or null) and the setter with be called here, than modifying on object passed as an argument